// yellowant-common-client/config/webpack.prod.js
const webpackMerge = require('webpack-merge');
const ExtractTextPlugin = require("extract-text-webpack-plugin");
const webpack = require("webpack");

const commonConfig = require('./webpack.common.js');
const helpers = require('./helpers.js');


const S3BUCKETPATH = process.env.S3BUCKETPATH || '/';

module.exports = function() {
  return webpackMerge(commonConfig(), {
    output: {
      publicPath: S3BUCKETPATH
    },
    module: {
      rules: [
        {
          test: /\.s?css$/,
          include: [ helpers.srcPath ],
          use: ExtractTextPlugin.extract({
            use: [{
              loader: "css-loader"
            }, {
              loader: "sass-loader"
            }],
            // use style-loader in development
            fallback: "style-loader"
          })
        }
      ]
    },
    plugins: [
      new ExtractTextPlugin({
        filename: "[name].css",
        disable: process.env.NODE_ENV === "development"
      }),
      new webpack.DefinePlugin({
        'process.env.S3BUCKETPATH': JSON.stringify(S3BUCKETPATH)
      })
    ]
  })
}
