// yellowant-common-client/config/webpack.dev.js
const webpackMerge = require('webpack-merge');
const path = require('path');
const HtmlWebpackPlugin  = require("html-webpack-plugin");

const commonConfig = require('./webpack.common.js');
const helpers = require('./helpers.js');

module.exports = function (env) {
  return webpackMerge(commonConfig(), {
    output: {
      publicPath: helpers.publicPath,
      crossOriginLoading: "anonymous",
    },

    module: {
      rules: [
        {
          test: /\.s?css$/,
          include: [ helpers.stylePath, helpers.angularPath, helpers.reactPath ],
          use: [{
              loader: "style-loader" // creates style nodes from JS strings
            }, {
              loader: "css-loader" // translates CSS into CommonJS
            }, {
              loader: "sass-loader" // compiles Sass to CSS
            }
          ]
        }
      ]
    },
    
    plugins: [
      new HtmlWebpackPlugin({
        filename: "index.html",
        template: path.resolve(helpers.reactPath, "GenericMessageRender", "App.html"),
        chunks: ["genericMessageRender"]
      })
    ],

    devtool: 'cheap-module-source-map',
    devServer: {
      headers: {
        'Access-Control-Allow-Origin': '*'
      },
      publicPath: helpers.publicPath,
      contentBase: path.resolve(helpers.reactPath, "GenericMessageRender"),
      hot: true,
      inline: true,
      port: 3000
    }
  })
}
