// yellowant-common-client/config/webpack.common.js
const path = require('path');
const BundleTracker = require('webpack-bundle-tracker');

const helpers = require('./helpers.js');

module.exports = function() {
  return {
    entry: {
      genericMessageRender: path.resolve(helpers.reactPath, "GenericMessageRender", "App.js"),
      developerToolbar: path.resolve(helpers.angularPath, "components", "developer-toolbar", "developer-toolbar.js"),
      marketplaceToolbar: path.resolve(helpers.angularPath, "components", "marketplace-toolbar", "marketplace-toolbar.js"),
      teamSwitcher: path.resolve(helpers.angularPath, "components", "team-switcher", "team-switcher.js")
    },
    output: {
      path: helpers.distPath,
      filename: '[name].js',
      sourceMapFilename: '[name].map'
    },

    externals: {
      'angular': 'angular',
      'angular-material': '"ngMaterial"',
      'angular-ui-router': '"ui.router"',
      'angular-messages': ' "ngMessages" ',
      'angular-aria': '"ngAria"'
    },

    resolve: {
      extensions: ['.js'],
      alias: {
        'react': 'react-lite',
        'react-dom': 'react-lite'
      }
    },
    module: {
      rules: [
        {
          test: /\.js$/,
          include: [ helpers.angularPath ],
          loader: 'babel-loader',
          options: {
            plugins: ['transform-runtime'],
            presets: [['env', { 'modules': false }], 'stage-0']
          }
        },
        {
          test: /\.js$/,
          include: [ helpers.reactPath ],
          loader: 'babel-loader',
          options: {
            plugins: ['transform-runtime'],
            presets: [['env', { 'modules': false }], 'react']
          }
        },

        {
          test: /\.html$/,
          include: [ helpers.angularPath ],
          use: [
            {
              loader: 'html-loader'
            }
          ]

        },

        {
          test: /\.(jpg|png|gif)$/,
          loader: 'file-loader',
          include: [ helpers.srcPath ],
          options: {
            outputPath: "assets/images/",
            name: '[name].[ext]'
          }
        }
      ]
    },

    plugins: [
      new BundleTracker({path: helpers.rootPath, filename: 'webpack-stats.json'})
    ]
  }
}
